x-pack/plugin/runtime-fields/src/test/java/org/elasticsearch/xpack/runtimefields/mapper/RuntimeScriptFieldMapperTests.java

```diff
@@ -311,110 +312,127 @@ private XContentBuilder mapping(String type, CheckedConsumer<XContentBuilder, IO
+                return (LongScriptFieldScript.Factory) (params, lookup) -> ctx -> new LongScriptFieldScript(params, lookup, ctx) {
+                    @Override
+                    public void execute() {
+                        new LongScriptFieldScript.Value(this).value(1);
                     }
-                    throw new IllegalArgumentException("No test script for [" + context + "]");
-                }
+                };
+            }
+            throw new IllegalArgumentException("No test script for [" + context + "]");
+        };
 
-                @Override
-                public Set<ScriptContext<?>> getSupportedContexts() {
-                    return Set.of(
-                        BooleanScriptFieldScript.CONTEXT,
-                        DateScriptFieldScript.CONTEXT,
-                        DoubleScriptFieldScript.CONTEXT,
-                        StringScriptFieldScript.CONTEXT,
-                        LongScriptFieldScript.CONTEXT
-                    );
-                }
-            };
+        @Override
+        public Set<ScriptContext<?>> getSupportedContexts() {
+            return Set.of(
+                BooleanScriptFieldScript.CONTEXT,
+                DateScriptFieldScript.CONTEXT,
+                DoubleScriptFieldScript.CONTEXT,
+                IpScriptFieldScript.CONTEXT,
+                StringScriptFieldScript.CONTEXT,
+                LongScriptFieldScript.CONTEXT
+            );
         }
+    };
+    private static final ScriptService SCRIPT_SERVICE = new ScriptService(
+        Settings.EMPTY,
+        Map.of(TEST_ENGINE.getType(), TEST_ENGINE),
+        TEST_ENGINE.getSupportedContexts().stream().collect(toMap(ctx -> ctx.name, Function.identity()))
+    );
+    private static final MapperRegistry MAPPER_REGISTRY = new IndicesModule(List.of(new RuntimeFields())).getMapperRegistry();
+
+    private MapperService createIndex(String index, Settings settings, XContentBuilder mappings) throws IOException {
+        IndexMetadata meta = IndexMetadata.builder("index")
+            .settings(Settings.builder().put("index.version.created", Version.CURRENT))
+            .numberOfReplicas(0)
+            .numberOfShards(1)
+            .build();
+        IndexSettings indexSettings = new IndexSettings(meta, Settings.EMPTY);
+        IndexAnalyzers indexAnalyzers = new IndexAnalyzers(
+            Map.of("default", new NamedAnalyzer("default", AnalyzerScope.INDEX, new StandardAnalyzer())),
+            Map.of(),
+            Map.of()
+        );
+        SimilarityService similarityService = new SimilarityService(indexSettings, SCRIPT_SERVICE, Map.of());
+        MapperService mapperService = new MapperService(
+            indexSettings,
+            indexAnalyzers,
+            xContentRegistry(),
+            similarityService,
+            MAPPER_REGISTRY,
+            () -> { throw new UnsupportedOperationException(); },
+            () -> true,
+            SCRIPT_SERVICE
```

Comment:
One thing I can't wrap my head around: as it stands this depends on #60933 . There is no final agreement yet on how to expose script service to the mappings code, so I was wondering whether it's wise to build on that, or whether we would still be able to adapt the test code once we find an alternative solution that we have agreement on.

Reply:
I think we can adapt it to whatever we decide on, yeah.